pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -104,9 +104,12 @@ func (f *frontend) _putOrPatchOpenShiftCluster(ctx context.Context, r *http.Requ
 	}
 
 	if !isCreate {
+		// Because the enricher updates doc in the DB,
+		// we must enrich it before setting non-terminal ProvisioningState,
+		// so the backend does not pick the doc up while frontend is still processing it
```

Comment:
I don't like it: it feels fragile. Maybe move db `Update` call out of `cachingEnricher` so we call it explicitly...